In [31]:
import pandas as pd
import numpy as np
import datetime
import math


Caching the list of root modules, please wait!
(This will only be done once - type '%rehashx' to reset cache!)



In [16]:
comments_df = pd.read_csv('../data/comments_df.csv')

In [17]:
comments_df.head(15)

Unnamed: 0             username         time_created  \
0            0            Tungsten_  2023-06-29 10:54:44   
1            1            Tungsten_  2023-06-30 11:33:11   
2            2       ProudBlackMatt  2023-06-29 11:16:15   
3            3         TomatoCanned  2023-06-29 11:09:47   
4            4         bad-fengshui  2023-06-29 11:26:41   
5            5      Pancake_muncher  2023-06-29 12:12:39   
6            6              suberry  2023-06-29 12:13:16   
7            7  Puzzled-Painter3301  2023-06-29 11:12:31   
8            8             e9967780  2023-06-29 11:55:52   
9            9                  NaN                  NaN   
10          10       OkartoIceCream  2023-06-29 11:50:34   
11          11        ShalomHasaeyo  2023-06-29 17:36:41   
12          12         SteadfastEnd  2023-06-29 11:02:09   
13          13           MsNewKicks  2023-06-29 19:26:21   
14          15              pal2002  2023-06-29 11:19:14   

                                                 body  \
0   [Megathread] Supreme Court Ruling on Affirmati...   
1   Thanks to everyone who engaged in insightful a...   
2   I would prefer using a process that takes into...   
3   u/Tungsten_, Thanks for creating a section jus...   
4   As with anything related to Asians in politics...   
5   Yet colleges will allow alumni and doners in e...   
6   I just hated Affirmative Action as a distracti...   
7   My own feeling is that I was never in love wit...   
8   Anti Asian racism whether against East Asians ...   
9   Can we overturn legacy and athlete admissions ...   
10  I want to remind people that in California, on...   
11  I think this is great news for our nation. Whi...   
12  My main concern is that, since the Court did n...   
13  There is obviously a lot to it and no real eas...   
14  I’m sure selective schools like Harvard will f...   

                                           tokens_new  \
0   ['Megathread', 'Supreme', 'Court', 'Ruling', '...   
1   ['Thanks', 'to', 'everyone', 'who', 'engaged',...   
2   ['I', 'would', 'prefer', 'using', 'a', 'proces...   
3   ['u/Tungsten_,', 'Thanks', 'for', 'creating', ...   
4   ['As', 'with', 'anything', 'related', 'to', 'A...   
5   ['Yet', 'colleges', 'will', 'allow', 'alumni',...   
6   ['I', 'just', 'hated', 'Affirmative', 'Action'...   
7   ['My', 'own', 'feeling', 'is', 'that', 'I', 'w...   
8   ['Anti', 'Asian', 'racism', 'whether', 'agains...   
9   ['Can', 'we', 'overturn', 'legacy', 'and', 'at...   
10  ['I', 'want', 'to', 'remind', 'people', 'that'...   
11  ['I', 'think', 'this', 'is', 'great', 'news', ...   
12  ['My', 'main', 'concern', 'is', 'that', 'since...   
13  ['There', 'is', 'obviously', 'a', 'lot', 'to',...   
14  ['I', '’m', 'sure', 'selective', 'schools', 'l...   

                                    normalized_tokens  \
0   ['megathread', 'supreme', 'court', 'ruling', '...   
1   ['thank', 'engage', 'insightful', 'respectful'...   
2   ['prefer', 'process', 'take', 'account', 'pove...   
3   ['u/tungsten_,', 'thank', 'create', 'section',...   
4   ['relate', 'asians', 'politic', 'm', 'see', 'l...   
5   ['college', 'allow', 'alumnus', 'doner', 'easi...   
6   ['hate', 'affirmative', 'action', 'distraction...   
7   ['feeling', 'love', 'affirmative', 'action', '...   
8   ['anti', 'asian', 'racism', 'east', 'asians', ...   
9   ['overturn', 'legacy', 'athlete', 'admission',...   
10  ['want', 'remind', 'people', 'california', 'pr...   
11  ['think', 'great', 'news', 'nation', 'abolish'...   
12  ['main', 'concern', 'court', 'strike', 'grutte...   
13  ['obviously', 'lot', 'real', 'easy', 'answer',...   
14  ['m', 'sure', 'selective', 'school', 'like', '...   

    normalized_tokens_count  word_count  
0                        62          78  
1                         9          20  
2                        52         103  
3                       126         269  
4                        25          59  
5                        19          40  
6         

## Divide comments and posts created pre-SCOTUS and post-SCOTUS ruling

In [18]:
ruling_date = datetime.datetime(2023, 6, 29)
print(ruling_date)

2023-06-29 00:00:00


In [20]:
first_row = comments_df.iloc[0]
first_time_str = first_row['time_created']

print(type(first_time_str))

first_datetime = datetime.datetime.fromisoformat(first_time_str)
print(type(first_datetime))

print(isinstance(first_time_str, str))

<class 'str'>
<class 'datetime.datetime'>
True


In [21]:
def str_to_datetime(str_):
    if isinstance(str_, str):
        return datetime.datetime.fromisoformat(str_)
    else:
        return None
        

comments_df['time_created_datetime'] = comments_df['time_created'].apply(lambda x: str_to_datetime(x))

In [34]:
comments_df.head(20)

Unnamed: 0             username         time_created  \
0            0            Tungsten_  2023-06-29 10:54:44   
1            1            Tungsten_  2023-06-30 11:33:11   
2            2       ProudBlackMatt  2023-06-29 11:16:15   
3            3         TomatoCanned  2023-06-29 11:09:47   
4            4         bad-fengshui  2023-06-29 11:26:41   
5            5      Pancake_muncher  2023-06-29 12:12:39   
6            6              suberry  2023-06-29 12:13:16   
7            7  Puzzled-Painter3301  2023-06-29 11:12:31   
8            8             e9967780  2023-06-29 11:55:52   
9            9                  NaN                  NaN   
10          10       OkartoIceCream  2023-06-29 11:50:34   
11          11        ShalomHasaeyo  2023-06-29 17:36:41   
12          12         SteadfastEnd  2023-06-29 11:02:09   
13          13           MsNewKicks  2023-06-29 19:26:21   
14          15              pal2002  2023-06-29 11:19:14   
15          16         Ok-Value5827  2023-06-29 13:12:38   
16          17              j3ychen  2023-06-29 17:49:07   
17          18     HappyPineapple11  2023-06-29 18:32:41   
18          19           alandizzle  2023-06-29 11:40:55   
19          20          gamesrgreat  2023-06-29 18:31:37   

                                                 body  \
0   [Megathread] Supreme Court Ruling on Affirmati...   
1   Thanks to everyone who engaged in insightful a...   
2   I would prefer using a process that takes into...   
3   u/Tungsten_, Thanks for creating a section jus...   
4   As with anything related to Asians in politics...   
5   Yet colleges will allow alumni and doners in e...   
6   I just hated Affirmative Action as a distracti...   
7   My own feeling is that I was never in love wit...   
8   Anti Asian racism whether against East Asians ...   
9   Can we overturn legacy and athlete admissions ...   
10  I want to remind people that in California, on...   
11  I think this is great news for our nation. Whi...   
12  My main concern is that, since the Court did n...   
13  There is obviously a lot to it and no real eas...   
14  I’m sure selective schools like Harvard will f...   
15  TBH, I don't know the depth of how affirmative...   
16  People can continue to debate the legal implic...   
17  The only issue with affirmative action was tha...   
18  You know what I hate?\n\nThe main subreddits w...   
19  I’m not a fan of affirmative action even tho i...   

                                           tokens_new  \
0   ['Megathread', 'Supreme', 'Court', 'Ruling', '...   
1   ['Thanks', 'to', 'everyone', 'who', 'engaged',...   
2   ['I', 'would', 'prefer', 'using', 'a', 'proces...   
3   ['u/Tungsten_,', 'Thanks', 'for', 'creating', ...   
4   ['As', 'with', 'anything', 'related', 'to', 'A...   
5   ['Yet', 'colleges', 'will', 'allow', 'alumni',...   
6   ['I', 'just', 'hated', 'Affirmative', 'Action'...   
7   ['My', 'own', 'feeling', 'is', 'that', 'I', 'w...   
8   ['Anti', 'Asian', 'racism', 'whether', 'agains...   
9   ['Can', 'we', 'overturn', 'legacy', 'and', 'at...   
10  ['I', 'want', 'to', 'remind', 'people', 'that'...   
11  ['I', 'think', 'this', 'is', 'great', 'news', ...   
12  ['My', 'main', 'concern', 'is', 'that', 'since...   
13  ['There', 'is', 'obviously', 'a', 'lot', 'to',...   
14  ['I', '’m', 'sure', 'selective', 'schools', 'l...   
15  ['TBH', 'I', 'do', "n't", 'know', 'the', 'dept...   
16  ['People', 'can', 'continue', 'to', 'debate', ...   
17  ['The', 'only', 'issue', 'with', 'affirmative'...   
18  ['You', 'know', 'what', 'I', 'hate', 'The', 'm...   
19  ['I', '’m', 'not', 'a', 'fan', 'of', 'affirmat...   

                                    normalized_tokens  \
0   ['megathread', 'supreme', 'court', 'ruling', '...   
1   ['thank', 'engage', 'insightful', 'respectful'...   
2   ['prefer', 'process', 'take', 'account', 'pove...   
3   ['u/tungsten_,', 'thank', 'create', 'section',...   
4   ['relate', 'asians', 'politic', 'm', 'see', 'l...   
5   [

In [23]:
first_datetime = comments_df.iloc[0]['time_created_datetime']
print(type(first_datetime))

<class 'pandas._libs.tslibs.timestamps.Timestamp'>


In [26]:
pre_scotus = comments_df[comments_df['time_created_datetime'] < datetime.datetime(2023, 6, 29)]

In [27]:
pre_scotus

Unnamed: 0              username         time_created  \
337          362      JesusSinfulHands  2020-07-13 11:22:51   
338          363           DJIisStupid  2020-07-13 12:38:51   
340          365      JesusSinfulHands  2020-07-13 11:26:06   
342          367                  ohea  2020-07-13 19:22:35   
343          368  Barebacking_Bernanke  2020-07-14 10:44:30   
...          ...                   ...                  ...   
3385        3726       notasinglesound  2019-04-24 16:21:43   
3386        3727            Marisa_Nya  2019-04-25 22:14:43   
3387        3728             lukarioDC  2019-04-23 23:43:15   
3388        3729            Marisa_Nya  2019-04-23 13:55:49   
3391        3732             arientyse  2019-05-29 14:15:14   

                                                   body  \
337   A big gap in support for affirmative action ha...   
338   wonder how this is split between income levels...   
340   Taken from this [story](http://aapidata.com/bl...   
342   Easy. It's overwhelmingly coming from people w...   
343   The decline neatly coincides with the effort t...   
...                                                 ...   
3385  Could you explain why/how AA was created for I...   
3386  Yeah. I definitely feel this. I'm Pakistani an...   
3387  For all those out there out-of-the-loop, here ...   
3388  I'm not sure why this reply is at -1. I'm tryi...   
3391  Please don't refer to us as "the blacks." It's...   

                                             tokens_new  \
337   ['A', 'big', 'gap', 'in', 'support', 'for', 'a...   
338   ['wonder', 'how', 'this', 'is', 'split', 'betw...   
340   ['Taken', 'from', 'this', 'story](http://aapid...   
342   ['Easy', 'It', "'s", 'overwhelmingly', 'coming...   
343   ['The', 'decline', 'neatly', 'coincides', 'wit...   
...                                                 ...   
3385  ['Could', 'you', 'explain', 'why', 'how', 'AA'...   
3386  ['Yeah', 'I', 'definitely', 'feel', 'this', 'I...   
3387  ['For', 'all', 'those', 'out', 'there', 'out',...   
3388  ['I', "'m", 'not', 'sure', 'why', 'this', 'rep...   
3391  ['Please', 'do', "n't", 'refer', 'to', 'us', '...   

                                      normalized_tokens  \
337   ['big', 'gap', 'support', 'affirmative', 'acti...   
338   ['wonder', 'split', 'income', 'level', 'migrat...   
340   ['take', 'story](http://aapidata.com', 'blog',...   
342   ['easy', 'overwhelmingly', 'come', 'people', '...   
343   ['decline', 'neatly', 'coincide', 'effort', 'd...   
...                                                 ...   
3385  ['explain', 'aa', 'create', 'ivy', 'league', '...   
3386  ['yeah', 'definitely', 'feel', 'm', 'pakistani...   
3387  ['loop', 'article](https://www.latimes.com', '...   
3388  ['m', 'sure', 'reply', 'm', 'try', 'objective'...   
3391  ['refer', 'black', 'okay', 'black', 'people', ...   

      normalized_tokens_count  word_count time_created_datetime  
337                        10          17   2020-07-13 11:22:51  
338                         6          11   2020-07-13 12:38:51  
340                        83         137   2020-07-13 11:26:06  
342                        72         152   2020-07-13 19:22:35  
343                        32          65   2020-07-14 10:44:30  
...                       ...         ...                   ...  
3385                       14          34   2019-04-24 16:21:43  
3386                       29          74   2019-04-25 22:14:43  
3387                       58         116   2019-04-23 23:43:15  
3388                        9          27   2019-04-23 13:55:49  
3391                        6          20   2019-05-29 14:15:14  

[2554 rows x 9 columns]

In [28]:
post_scotus = comments_df[comments_df['time_created_datetime'] >= datetime.datetime(2023, 6, 29)]

In [29]:
post_scotus

Unnamed: 0              username         time_created  \
0              0             Tungsten_  2023-06-29 10:54:44   
1              1             Tungsten_  2023-06-30 11:33:11   
2              2        ProudBlackMatt  2023-06-29 11:16:15   
3              3          TomatoCanned  2023-06-29 11:09:47   
4              4          bad-fengshui  2023-06-29 11:26:41   
...          ...                   ...                  ...   
333          357              e9967780  2023-06-29 12:23:45   
334          358    Different-Rip-2787  2023-06-29 18:03:45   
336          361    Different-Rip-2787  2023-06-29 19:16:06   
3365        3705   DudeGuyBroPalBuddie  2024-03-14 09:02:59   
3366        3706  Extension_River_9901  2024-04-12 11:37:00   

                                                   body  \
0     [Megathread] Supreme Court Ruling on Affirmati...   
1     Thanks to everyone who engaged in insightful a...   
2     I would prefer using a process that takes into...   
3     u/Tungsten_, Thanks for creating a section jus...   
4     As with anything related to Asians in politics...   
...                                                 ...   
333   You should ask this question from a student wh...   
334   I fail to see how this is 'detrimental' to any...   
336   > Those same options were also available to As...   
3365  How White Supremacy uses Taiwan This is a long...   
3366  White supremacy wants less Asian people in the...   

                                             tokens_new  \
0     ['Megathread', 'Supreme', 'Court', 'Ruling', '...   
1     ['Thanks', 'to', 'everyone', 'who', 'engaged',...   
2     ['I', 'would', 'prefer', 'using', 'a', 'proces...   
3     ['u/Tungsten_,', 'Thanks', 'for', 'creating', ...   
4     ['As', 'with', 'anything', 'related', 'to', 'A...   
...                                                 ...   
333   ['You', 'should', 'ask', 'this', 'question', '...   
334   ['I', 'fail', 'to', 'see', 'how', 'this', 'is'...   
336   ['>', 'Those', 'same', 'options', 'were', 'als...   
3365  ['How', 'White', 'Supremacy', 'uses', 'Taiwan'...   
3366  ['White', 'supremacy', 'wants', 'less', 'Asian...   

                                      normalized_tokens  \
0     ['megathread', 'supreme', 'court', 'ruling', '...   
1     ['thank', 'engage', 'insightful', 'respectful'...   
2     ['prefer', 'process', 'take', 'account', 'pove...   
3     ['u/tungsten_,', 'thank', 'create', 'section',...   
4     ['relate', 'asians', 'politic', 'm', 'see', 'l...   
...                                                 ...   
333   ['ask', 'question', 'student', 'fcuke', 'sacri...   
334   ['fail', 'detrimental', 'black', 'brown', 'stu...   
336   ['>', 'option', 'available', 'asam', 'student'...   
3365  ['white', 'supremacy', 'use', 'taiwan', 'long'...   
3366  ['white', 'supremacy', 'want', 'asian', 'peopl...   

      normalized_tokens_count  word_count time_created_datetime  
0                          62          78   2023-06-29 10:54:44  
1                           9          20   2023-06-30 11:33:11  
2                          52         103   2023-06-29 11:16:15  
3                         126         269   2023-06-29 11:09:47  
4                          25          59   2023-06-29 11:26:41  
...                       ...         ...                   ...  
333                         8          23   2023-06-29 12:23:45  
334                        37          88   2023-06-29 18:03:45  
336                        31          70   2023-06-29 19:16:06  
3365                     1261        2644   2024-03-14 09:02:59  
3366                       28          49   2024-04-12 11:37:00  

[304 rows x 9 columns]

In [36]:
no_datetime = comments_df[pd.isna(comments_df['time_created'])]\

print(no_datetime.shape)

(534, 9)


Breakdown of datetimes:
- Before SCOTUS decision: 2554 rows
- After SCOTUS decision: 304 rows

- time_created is NaN: 534 rows